### PR TITLE
emscripten: 2.0.1 -> 2.0.10

### PR DESCRIPTION
**Pull request by athas**
binaryen: 96 -> 99

These updates must be in the same commit because emscripten depends on
a specific version of binaryen.

###### Motivation for this change

Emscripten was old.  Now it is slightly less old.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by athas** [comment]
Note that this isn't the *newest* Emscripten, but it's the newest I could get to work (see #130659 for my other attempt).

File: pkgs/development/compilers/binaryen/default.nix [hunk 1]
```diff
@@ -2,23 +2,15 @@
...
     owner = "WebAssembly";
     repo = "binaryen";
     rev = "version_${version}";
-    sha256 = "1mqpb6yy87aifpbcy0lczi3bp6kddrwi6d0g6lrhjrdxx2kvbdag";
+    sha256 = "1a6ixxm1f8mrr9mn6a0pimajdzsdr4w1qhr92skxq67168vvc1ic";
   };
 
-  patches = [
-    # Adds --minimize-wasm-changes option required by emscripten 2.0.1
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/binaryen/default.nix:16]: Please remove the unused fetchpatch from inputs.

Comment (athas) [reply pkgs/development/compilers/binaryen/default.nix:16]: Done.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130723` at a9d3d2f3 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/caced786d2c060df79d91d317edaea72">foo-yc20</a></li>
    <li><a href="https://gist.github.com/2ec2d1e104bf410e0f12885ba261b98a">kapitonov-plugins-pack</a></li>
    <li><a href="https://gist.github.com/313f9f1a28f4341a6512a68555483ea4">purePackages.faust</a></li>
  </ul>
</details>
<details>
  <summary>14 packages built successfully:</summary>
  <ul>
    <li>binaryen</li>
    <li>emscripten</li>
    <li>faust (faust2)</li>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2csound</li>
    <li>faust2firefox</li>
    <li>faust2jack</li>
    <li>faust2jackrust</li>
    <li>faust2jaqt</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by asppsa — APPROVED** [review]

**Pull request merged**